src/read.jl

```diff
@@ -181,12 +198,13 @@ data from `io` into the returned `EDF.File`, call `EDF.read!(file)`.
 """
 function File(io::IO)
     file_header, signal_count = read_file_header(io)
+    T = _sample_type(file_header)
```

Comment:
It occurred to me that we might actually want to make this a type parameter of `EDF.File` itself since the sample type is file-global; you can't have 16- and 24-bit values in the same file (AFAIK). Then the signals vector will be `Union{AnnotationsSignal,Signal{T}}` to ensure this rather than `Union{AnnotationsSignal,Signal}` which will allow mixing representations.

Reply:
Uh, I need to check what GDF and EDF+ do

Reply:
Okay, EDF+ doesn't seem to allow it, nor does [GDF](https://arxiv.org/abs/cs/0608052).

Reply:
Well actually I guess we can make it a union with `Signal{T}` without introducing a type parameter on the `File` type

Reply:
But the more I think about this, the more I think it should be on the `File` type as well because it really does determine the filetype/format as much / more than it determines the signal type.

Reply:
Implemented in the most recent commit.